client.go

```diff
@@ -77,9 +86,15 @@ func (c *Client) StartFlushInterval(interval time.Duration) {
 	}()
 }
 
+func (c *Client) sendSynchronously() {
+	c.buffer.Flush()
+}
+
 func (c *Client) StopFlushInterval() {
-	c.ticker.Stop()
-	c.tickerDone <- true
+	if c.buffer.autoFlush {
```

Comment:
There's no need to check against the flush config, we should however validate that the ticker isn't `nil` to prevent a panic:

```suggestion
	if c.ticker != nil {
```